### PR TITLE
Fixes #35888 - Full host image fails with bad request

### PR DESCRIPTION
**Pull request by stejskalleos**
Enable https connection when uri scheme is https

### CODE REVIEW

File: app/services/foreman_bootdisk/iso_generator.rb [hunk 1]
```diff
@@ -217,25 +217,22 @@ def self.fetch(path, uri, limit = 10)
...
-                raise("Too many HTTP redirects when downloading #{uri}") if limit <= 0
-
-                fetch(path, response['location'], limit - 1)
-                # prevent multiple writes to the cache
-                write_cache = false
-              else
-                raise ::Foreman::Exception, N_(format("Unable to download boot file %{uri}, HTTP return code %{code}", uri: uri, code: response.code))
-              end
+
+          response = http_object.get_response(uri)
```
Comment (ares) [inline app/services/foreman_bootdisk/iso_generator.rb:221]: the `Accept-Encoding` was added for this reason https://projects.theforeman.org/issues/25992, does it now work even without?

File: app/services/foreman_bootdisk/iso_generator.rb [hunk 2]
```diff
@@ -217,25 +217,22 @@ def self.fetch(path, uri, limit = 10)
...
+
+          case response
+          when Net::HTTPSuccess then
+            response.body do |chunk|
```
Comment (ares) [inline app/services/foreman_bootdisk/iso_generator.rb:225]: does body actually accept a block? the docs says differently, internally it calls read_body without passing a block or any argument

**Review by ares — CHANGES_REQUESTED** [review]
I'm not necessarily requesting changes, just answers :-)

**Comment by stejskalleos** [comment]
Found the issue, problem was in missing `use_ssl=true` for `uri` with `https` scheme

**Review by ares — APPROVED** [review]
Thanks, fixes the issue. Merging!

**Pull request merged**